packages/dd-trace/src/log/log.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const { format } = require('util')
+
+class Log {
+  constructor (message, args, cause, delegate) {
+    this.message = message
+    this.args = args
+    this.cause = cause
+    this.delegate = delegate
+  }
+
+  get formatted () {
+    const { message, args } = this
+
+    let formatted = message
+    if (message && args?.length) {
+      formatted = format(message, ...args)
+    }
+    return formatted
+  }
+
+  static parse (...args) {
+    let message, cause, delegate
+
+    const lastArg = args[args.length - 1]
+    if (lastArg && typeof lastArg === 'object' && lastArg.stack) { // lastArg instanceof Error?
```

Comment:
could it check directly if `lastArg` is an instance of `Error`? or should we continue supporting an object with `stack` property?

Reply:
I think we've had cases where it looked like an error but didn't extend from `Error`, especially from auto-instrumentation where we log errors from external libraries automatically. We could revisit it but it's out of scope of this PR.